components/launcher/src/service.rs

```diff
@@ -89,8 +90,15 @@ fn pipe_stdout<T>(out: T, id: &str)
     let mut reader = BufReader::new(out);
     let mut buffer = String::new();
     while reader.read_line(&mut buffer).unwrap() > 0 {
-        let line = output_format!(preamble &id, logkey "O", buffer);
-        writeln!(&mut io::stdout(), "{}", line).expect("unable to write to stdout");
+        let so = StructuredOutput::new(&id,
+                                       "O",
+                                       line!(),
+                                       file!(),
+                                       column!(),
+                                       output::get_format(),
+                                       output::get_verbosity(),
+                                       &buffer.trim_right_matches('\n'));
```

Comment:
I thought we had macros that wrapped all the verbosity of creating these `StructuredOutput` structs?

Reply:
these 2 calls in the launcher used a macro `output_format!` which wrapped the `StructuredOutput`. They were the only callers. That particular macro was problematic in that it just formats the output and returns a string that the launcher renders here to either `stdout` or `stderr`. That works great in an `ANSI` terminal world but it breaks down when you need to make system API calls when ever you need to change color or formatting.

We sould probably have the `stdout` call leverage `outputln!` but I thought it seemed strange to have one call a macro and the other deal directly with `StructuredOutput` and it seemed overboard to create a set of corresponding `outputerrln!` macros for only this one caller.

Hope that makes sense but totally open to suggestions.